Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -427,60 +421,134 @@ function Unregister-ServiceFabricApplicationTypeAction
         $TimeoutSec
     )
 
-    $global:operationId = $SF_Operations.UnregisterApplicationType
+    $global:operationId = $SF_Operations.GetApplicationType
+    $getAppTypeAction = { Get-ServiceFabricApplicationType -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion }
+    $getAppTypeRetryEvaluator = {
+        param($appType)
 
-    $unregisterAction = {
-        Unregister-ServiceFabricApplicationType -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion -Force -TimeoutSec $TimeoutSec
-        if (!$?)
+        # If provisioning not started, retry register
```

Comment:
Since this is wait, you should throw if appType is null

Reply:
As discussed, this is correct behavior.